### ISSUE TITLE
New Hyperopt tutorial for dealing with hyperparameter spaces

### ISSUE THREAD

**Issue by guillaume-chevalier** [hyperopt/hyperopt#316]
Hi @jaberg,

I've been interested in Hyperopt and I've found it quite useful. I have then written an Hyperopt tutorial to introduce how to define hyperparameter spaces with a dict and how to use fmin with trials to keep track of progress and slay down the search space.

From what I see, it is said here to open an issue if anyone has suggestions for listing related work: https://github.com/hyperopt/hyperopt/wiki/RelatedWork

Here is my tutorial's official link: http://vooban.com/en/tips-articles-geek-stuff/hyperopt-tutorial-for-optimizing-neural-networks-hyperparameters/

And here is a GitHub repository containing the tutorial's python notebook file as well as a project built with Hyperopt for giving a full (and simple) implementation example: https://github.com/Vooban/Hyperopt-Keras-CNN-CIFAR-100

To sum up, the tutorial is especially built to learn how to use and manipulate hyperopt's hyperspaces with dicts, nested choices, and on. Feel free to link to that in your wiki or elsewhere in your documentation if you think that can be helpful.

Thanks!

**Comment by ivan-marroquin** [comment]
Hi  Guillaume,

Many thanks for the tutorial. I would like to ask you a question with respect to the hyperparameter space configuration.

I want to minimize a function, and for which I have initial conditions values. My idea is to include these conditional values into the hyperparameter space. So then, I would like to use these values during the first trial of the fmin function call. Thereafter, the initial condition values are not used again during the rest of iterations of the fmin function.

Here is part of the Python code:
a) define the hyperparameter space, whose first value correspond to the initial conditions:
space_1= np.insert(np.random.normal(np.mean(xtest_data[:,0]), np.std(xtest_data[:,0]), 96), 0, logs_litho1_max[0])
space_2= np.insert(np.random.normal(np.mean(xtest_data[:,1]), np.std(xtest_data[:,1]), 96), 0, logs_litho1_max[1])
space_3= np.insert(np.random.normal(np.mean(xtest_data[:,2]), np.std(xtest_data[:,2]), 96), 0, logs_litho1_max[2])
space_4= np.insert(np.random.normal(np.mean(xtest_data[:,3]), np.std(xtest_data[:,3]), 96), 0, logs_litho1_max[3])

b) define the search space:
parameter_space_litho1= {
        'SP_log': hp.choice('SP_log', space_1),
        'RT_log': hp.choice('RT_log', space_2),
        'RHOB_log': hp.choice('RHOB_log', space_3),
        'NPSS_log': hp.choice('NPSS_log', space_4),
       }

c) function to minimize:
def optimizer_litho1(parameters):
    print("Parameters values used for the optimization:\n")
    pprint(parameters)
    print("\n")
    
    n_sp_log= np.round(parameters['SP_log'], 3)
    n_rt_log= np.round(parameters['RT_log'], 3)
    n_rhob_log= np.round(parameters['RHOB_log'], 3)
    n_npss_log= np.round(parameters['NPSS_log'], 3)

   function_to_minimize= .....
  ......
   return loss

In section (c) is where I want to make sure that the initial conditions are only used during the first trial. For the rest of iterations, the code should use any other combinations for the variables used to minimize the function.

Many thanks,

Ivan
